src/crd-controller/HostedServices/V1AlphaController.cs

```diff
@@ -155,8 +158,15 @@ private async Task HandleAdd(KamusSecret kamusSecret)
                 Type = kamusSecret.Type,
                 StringData = decryptedItems
             };
-
-            await mKubernetes.CreateNamespacedSecretAsync(secret, @namespace);
+            
+            if (isUpdate)
```

Comment:
This is a bit ugly, I would consider moving the common part to a method (L 128-160) and have 2 separate methods.